### PR TITLE
Refs #67: initial code for changes

### PR DESCRIPTION
**Pull request by jpollock**

### CODE REVIEW

File: cli/app/api.go [hunk 1]
```diff
@@ -442,45 +443,520 @@ func PublishToMashery(user *ApiUser, appDir string, gatewayJSON string, host str
...
 	}
 
-	return nil
+	return apiId, apiName, endpoints, updated
+}
+
+func merge(dst, src map[string]interface{}, depth int) map[string]interface{} {
+	for key, srcVal := range src {
+		if dstVal, ok := dst[key]; ok {
+			if reflect.ValueOf(dstVal).Kind() == reflect.Map {
```
Comment (pointlander) [inline cli/app/api.go]: Would a type switch work here instead of reflection: https://tour.golang.org/methods/16

File: cli/app/api.go [hunk 2]
```diff
@@ -442,45 +443,520 @@ func PublishToMashery(user *ApiUser, appDir string, gatewayJSON string, host str
...
+		if i_d["requestPathAlias"] == ep["requestPathAlias"] {
+			return true
+		}
+	}
+	return false
+}
+func mapify(i interface{}) (map[string]interface{}, bool) {
+	value := reflect.ValueOf(i)
+	if value.Kind() == reflect.Map {
+		m := map[string]interface{}{}
```
Comment (pointlander) [inline cli/app/api.go]: Would a type assertion work here instead of reflection: https://tour.golang.org/methods/15

File: cli/app/api.go [hunk 3]
```diff
@@ -442,45 +443,520 @@ func PublishToMashery(user *ApiUser, appDir string, gatewayJSON string, host str
...
+	}
+}
+
+func CreateOrUpdatePackage(user *ApiUser, token string, packagePlanDoc []byte, apiName string, updated bool) string {
+	var p string
+	if !updated {
+		p, err := user.Create("packages", "id,name,plans.id,plans.name", string(packagePlanDoc), token)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: Unable to create the package %s\n\n", p)
+			fmt.Errorf("%v", err)
```
Comment (pointlander) [inline cli/app/api.go]: This statement is a noop

File: cli/app/api.go [hunk 4]
```diff
@@ -442,45 +443,520 @@ func PublishToMashery(user *ApiUser, appDir string, gatewayJSON string, host str
...
+		merged := merge(m, m1, 0)
+		var mergedDoc []byte
+		if mergedDoc, err = json.Marshal(merged); err != nil {
+			panic(err)
+		}
+		packageId := merged["id"].(string)
+		p, err = user.Update("packages/"+packageId, "id,name,plans.id,plans.name", string(mergedDoc), token)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: Unable to update the package %s\n\n", p)
+			fmt.Errorf("%v", err)
```
Comment (pointlander) [inline cli/app/api.go]: This statement is a noop

**Comment by galois17** [comment]
@jpark800 I have squashed/rebased this branch against master into 1 commit and resolved conflicts.

**Pull request merged**